### DISCUSSION TITLE
Using Custom Style Property

### DISCUSSION THREAD

**Discussion by RakulAgn** [GrapesJS/grapesjs#5714] [General]
Is there any options is there to access the domComponent in emit function it have only the updateStyle function 

<img width="1049" alt="image" src="https://github.com/GrapesJS/grapesjs/assets/119593989/395ac388-2475-403b-9dab-be067b32049a">

I have Some usecase for aligning image like left , center, right and also it have property full when the value is full i need to set the image width 100%. this is want in above image but see that i need to access the component through the getSelectedComponent and i need to update those changes. instead of that if i can access the component directly through emit function it could be better right ?

**Comment by artf** [comment]
sorry can you elaborate better on what exactly you need?

**Reply by RakulAgn** [reply]
Sure, what I'm attempting to accomplish with the Style Manager is the creation of a custom alignment for an image block. The alignment options include Left, Center, Right, which trigger text-align to adjust the image's position within the div (left, right, center), and a Full property. This Full property is designed to resize the image according to the parent container's dimensions, effectively merging two styling properties into a singular, stylable approach, as demonstrated in the screenshot provided.

![image](https://github.com/GrapesJS/grapesjs/assets/119593989/a15b737d-b0fd-40c9-9c9a-71ec96669b98)

The main issue arises when implementing the functionality for the Full property. I'm unable to easily access the child component (the image within the container). Ideally, I need to select the current component, verify if it contains an image child, and then apply the necessary functionality. However, this approach is not straightforward when working within the Style Manager. If the emit method could return a component, it would facilitate the creation of a more customized Style Manager. Currently, I'm finding it challenging to achieve a level of customization similar to platforms like Brevo and Klaviyo due to these limitations.

**Reply by artf** [reply]
> If the emit method could return a component, it would facilitate the creation of a more customized Style Manager

Can you explain what kind of difference you think to have from component passed from emit to the one you already using via `editor.getSelected()`? It would be exactly the same thing.

**Reply by RakulAgn** [reply]
By Default the Style Manager Targets the current selected component but it that case it should also return child components too ( as like in trait )  if its return those things it could be help full do get the relevant component. so basically this is skipping additional one or two lines. i think that might be a difference right ?

**Reply by artf** [reply]
> the Style Manager Targets the current selected component 

No, it mainly targets the CSSRule, not the Component, so it's quite different from the Traits.

> if its return those things it could be help full do get the relevant component. so basically this is skipping additional one or two lines. i think that might be a difference right

Again, no difference at all, you would just skip the `editor.getSelected()` line.

I'd suggest you learn how to write DRY code though, that would definitely help reduce all those lines of code in your snippet + read the [documentation](https://grapesjs.com/docs/api/component.html) you might find some useful methods on how to find inner components without all those ugly loops.

**Reply by RakulAgn** [reply]
Yeah Sure
